torch/distributed/pipelining/stage.py

```diff
@@ -594,6 +615,24 @@ def backward_one_chunk(
         self.grads_input = self.backward_maybe_with_nosync(bwd_kwargs)
```

Comment:
`backward_maybe_with_nosync` needs to be updated right?

Reply:
in the future, we can add a PR which lets us build our own dw_runner without the user providing it.  If we do that, we'd have to modify `backward_maybe_with_nosync` to only run part of backward and then defer the rest into the generated runner function we'd call later.

however in this PR, we do not provide such automation.  We require the user to already modify their backward to separate out the parts that are done during .backward (B) and the parts done in dw_runner (W).  Hence, for now we can just always call `backward_maybe_with_nosync`.   IIUC.